### PR TITLE
filter duplicate map search results

### PR DESCRIPTION
**Pull request by wginsberg**
part of Couchers-org/couchers#4132

**Web frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [x] There are no console warnings when running the app
- [x] Added any new components to storybook
- [x] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes

## before
Screen Shot 2022-02-10 at 11 22 55 AM

## after
Screen Shot 2022-02-10 at 11 22 48 AM

### CODE REVIEW

File: app/web/utils/nominatim.ts [hunk 1]
```diff
@@ -42,3 +42,21 @@ export const simplifyPlaceDisplayName = (place: NominatimPlace) => {
   }
   return addressParts.join(", ");
 };
+
+export const filterDuplicatePlaces = (places: NominatimPlace[] = []) => {
+  const placeImportance = places.reduce((places, place) => {
```
Comment (lucaslcode) [inline app/web/utils/nominatim.ts]: This naming is a tiny bit confusing:
- `places` is shadowed
- `placeImportance` sounds like a `Record<string, number>`. Maybe it should be `deduplicatedPlaces`

**Review by lucaslcode — CHANGES_REQUESTED** [review]
I could be wrong, but [looking here](https://nominatim.openstreetmap.org/ui/search.html?q=burlington&addressdetails=1), all the dupes are because of one result for the city, and another for the administrative boundary?
Maybe we should eliminate boundaries instead?

**Comment by wginsberg** [comment]
@lucaslcode 

In other cases there is no "city" result at all, just two "administrative" results. Working with addresses is fun, eh!

https://nominatim.openstreetmap.org/ui/search.html?q=toronto

File: app/web/utils/nominatim.ts [hunk 2]
```diff
@@ -42,3 +42,21 @@ export const simplifyPlaceDisplayName = (place: NominatimPlace) => {
...
+
+    return previousRecord[displayName]?.importance >= importance
+      ? previousRecord
+      : { ...previousRecord, [displayName]: currentPlace };
+  }, {} as Record<string, NominatimPlace>);
+
+  const filtered = places.filter((place) => {
+    const displayName = simplifyPlaceDisplayName(place);
+    return deduplicatedPlaces[displayName].place_id === place.place_id;
+  });
```
Comment (darrenvong) [inline app/web/utils/nominatim.ts]: Could this bit not be done as part of the `reduce` too? Seems unnecessary to loop through `places` again when it's already doing some work there.

Comment (wginsberg) [reply app/web/utils/nominatim.ts]: I was thinking this was needed to preserve ordering but actually `Object.values(deduplicatedPlaces)` does that too. will replace it with that

**Review by lucaslcode — APPROVED** [review]
Thanks so much, and for the great tests!

**Pull request merged**